docs/DeveloperGuide.md

```diff
@@ -2,37 +2,159 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The CS2113 Team (Professor Akshay <3) for the guidance and teaching us concepts of Software Engineering and OOP
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Main overview of the system:
 
+* Main Components
+  * Ui: Deals with user input/output
+  * Storage: Deals with data management on hard disk
+  * Parser: Deals with parsing user input
+  * Command: Deals with handling command input logic
+
+![broadOverview.png](UML%2FImage%2FbroadOverview.png)
+
+Fig 1
+
+
+### Ui 
+
+Ui class involves all the methods that display messages to the user when the user is interacting with the porgram.
+
+Types of methods:
+
+1. Greeting Messages: Messages shown to welcome user and acknowledge their exit
+2. Confirmation Messages: Messages shown to inform user that their desired action has been carried out
+3. Exception Messages: Messages shown to inform user of their invalid actions and its reasons
+
+![img_2.png](img_2.png)
+
+Fig 2
+
+
+### Storage
+
+
+<code>sequence diagram</code>
+
+### Parser
+
+Parser involves dealing with parsing user input to understand what the user wants to do. It returns a 
+<code>Command</code> class that it executed during the run-time.
+
+### Command
+
+Commands are the way we execute the programme features, based on parsed user input
+
+<code>List of commands</code>
+
+<code> Seq of 1 command</code>
+
+### [DONE] Duplication checker feature
+#### Implementation
+The proposed <code>duplicationChecker</code> is facilitated by <code>Parser</code>. It will check against the existing
```

Comment:
All of the developer guide is kind of hard to read. Maybe increasing font size or design could lead to more readability.